### DISCUSSION TITLE
Option for notification after X fails

### DISCUSSION THREAD

**Discussion by Bitals** [borgbase/vorta#1779] [Ideas]
Hello, and thank you very much for a great utility that I use daily and that saved me from headaches multiple times over the years.

I am thinking about implementing a feature that sends a notification only after X (by default 3, for example, but could be tweaked) failed backup attempts in a row.

The reason being that there may be a lot of "false-positive" fails when you schedule backups often, like when waking up my laptop I sometimes receive 1 or 2 notifications about fails in the first seconds before it even reconnects to wi-fi, or they are already there when I open the lid. So if a user creates backups like every hour, they will most likely be fine with 2 failed attempts, as long as the 3rd one is successful.
Another example is a known unstable network. The user might want to know about every single failed backup, but it also might become uninformative and annoying if it is spammed on them. Ultimately the choice is theirs.

I see this as a completely optional feature, not affecting the defaults and the regular "Notify on **every** fail" setting, but rather being it's subsidiary.

So this is my suggestion/contribution offer.
Although I am not very experienced with Python development, and it may take me some time, but it looks like a good learning experience.

**Comment by real-yfprojects** [comment]
I see how this feature would be useful but I have the concern that too many options clutter Vorta's UI too much.

**Comment by oliver124810** [comment]
This option would be very useful. With a laptop, the false alarms are really annoying. Disabling the notifications is not a good workaround unfortunately. I understand the concern about cluttering the UI. On the other hand, not having the option makes the actual feature useless, since with so many false positives, one starts to ignore the notifications altogether. To put it differently, on a laptop, this is not an option for the feature, but a must :)

In terms of usability, it would better though to allow the notification after a period of time, e.g. one day. The UI would be very similar to the one of the backup schedule interval, and make it easy for users to understand. An alternative would be to make the time fixed. This is how Apple does it in TimeMachine (search for "time machine no backup notification" for examples). The delay is not configurable AFAIK. Maybe that would be an option too, to avoid cluttering the UI, i.e. notify only after 12h or 24h.

**Comment by oliver124810** [comment]
On a second thought, there is an additional, related use case: sometimes I have to disable the scheduling, for instance because the network is slow, unreliable or metered, or because I need the full bandwidth for other applications or speed tests. It is easy to forget to enable the scheduling again.

An option like "Notify if no backup in the last X days" would cover this use case, and also when all recent backup runs failed. It would be also nearer to the user domain.

**Reply by real-yfprojects** [reply]
Internally we already have the option to pause automatic backups for a specifyed time interval. Wouldn't that fit better for this use case?

**Reply by oliver124810** [reply]
Yes, that would also help with the the use case, but I would not say it is always better. I need this for instance when I travel with my laptop and I have only sporadic access to the internet. If I had to pause the backup, I would not know for how long. A warning that I did not backup for N days would be better, because I would start to seek for an internet access. Also, if I paused for four days, it would not back up if I am online after two days, which is unwanted.
